app/src/main/java/com/mapbox/vision/examples/activity/main/MainActivity.kt

```diff
@@ -475,74 +517,42 @@ class MainActivity : AppCompatActivity() {
         lines_detections_container.hide()
     }
 
-    private fun setSignClassificationMode() {
+    private fun setAppMode(mode: AppMode) {
+        resetAverageFPS()
         soundsPlayer.stop()
 
-        vision_view.visualizationMode = VisualizationMode.Clear
-        appMode = AppMode.Classification
-
-        tracker = Tracker(5)
-
-        hideLineDetectionContainer()
-        sign_info_container.show()
-        dashboard_container.hide()
-        safety_mode_container.hide()
         back.show()
-    }
-
-    private fun setDetectionMode() {
-        soundsPlayer.stop()
-
-        vision_view.visualizationMode = VisualizationMode.Detections
-        appMode = AppMode.Detection
-
-        hideLineDetectionContainer()
-        hideSignsContainer()
         dashboard_container.hide()
         safety_mode_container.hide()
-        back.show()
-    }
-
-    private fun setSegmentationMode() {
-        soundsPlayer.stop()
-
-        vision_view.visualizationMode = VisualizationMode.Segmentation
-        appMode = AppMode.Segmentation
-
-        hideLineDetectionContainer()
-        hideSignsContainer()
-        dashboard_container.hide()
-        safety_mode_container.hide()
-        back.show()
-    }
-
-    private fun setSafetyMode() {
-        soundsPlayer.stop()
-
-        vision_view.visualizationMode = VisualizationMode.Clear
-        appMode = AppMode.Safety
-
         hideLineDetectionContainer()
         hideSignsContainer()
-        dashboard_container.hide()
-        safety_mode_container.show()
-        calibration_progress.hide()
-        distance_to_car_label.hide()
-        safety_mode.hide()
-        back.show()
-    }
 
-    private fun setLaneDetectionMode() {
-        soundsPlayer.stop()
-
-        vision_view.visualizationMode = VisualizationMode.Clear
-        appMode = AppMode.Lanes
+        appMode = mode
+        when (appMode) {
+            AppMode.Classification -> {
+                vision_view.visualizationMode = VisualizationMode.Clear
+                tracker = Tracker(5)
+                sign_info_container.show()
+            }
+            AppMode.Detection -> {
+                vision_view.visualizationMode = VisualizationMode.Detections
+                lines_detections_container.show()
```

Comment:
Do we really need line detections here? Seems they should be shown on `Lanes` mode that shows lanes detected